### PR TITLE
Implemented a decent amount of the translation use case [See desc]

### PR DESCRIPTION
**Pull request by jaronfernandes**
Completed a sizeable amount of the translation use-case, with the exception of:
- Hooking use case up with the View
- Extensive testing with the API, trying to find edge cases
- Displaying it to the View.

### CODE REVIEW

**Review by Jaiz183 — CHANGES_REQUESTED** [review]
I think you shouldn't merge your branch right now. If you need to make smaller features, make sub-branches. As we learnt recently, pull requests are primarily made when you complete a feature, not when you are in an intermediate stage. When you're ready with the translation use case, I'll approve the PR. Regardless, your changes look good so far, apart from some violated coding conventions (Javadocs, try-catch statements). Good attention to detail to catch mistakes like non-private fields. Sufficient description.

**Comment by jaronfernandes** [comment]
> I think you shouldn't merge your branch right now. If you need to make smaller features, make sub-branches.

Gotcha, that's a good idea. I'll create a PR when the whole use-case is ready.

File: src/app/TranslationUseCaseFactory.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import interface_adapter.translation.TranslationController;
+import interface_adapter.translation.TranslationPresenter;
+import interface_adapter.translation.TranslationPresenter;
+import use_case.translation.TranslateAPIDataAccessInterface;
+import use_case.translation.TranslationInputBoundary;
+import use_case.translation.TranslationInteractor;
+import use_case.translation.TranslationOutputBoundary;
+import view.ArticleView;
+
+public class TranslationUseCaseFactory {
```
Comment (Jaiz183) [inline src/app/TranslationUseCaseFactory.java:14]: You should write Javadocs for this class.

File: src/data_access/APIDataAccessObject.java [hunk 2]
```diff
@@ -71,16 +84,76 @@ public List<Article> getArticles(String rawSearchTerm) {
...
+
+    @Override
+    public TranslatedArticle translateArticle(Article article, String language) {
+        TranslatedArticleFactory translatedArticleFactory = new TranslatedArticleFactory();
+        String translatedText = null;
+
+        // sample: https://api-free.deepl.com/v2/translate?auth_key=
+        // 8dbcc2f3-03ef-8e22-3c4a-718f08bbe557:fx
+        // &text=This%20is%20a%20Test.&target_lang=JA
+        try {
```
Comment (Jaiz183) [inline src/data_access/APIDataAccessObject.java:126]: You should use try-catch statements around smaller blocks of code. This statement could catch errors in deserialisation, decoding or request sending.

File: src/view/HomeView.java [hunk 3]
```diff
@@ -1,26 +1,22 @@
...
 import java.beans.PropertyChangeEvent;
 import java.beans.PropertyChangeListener;
-import java.util.ArrayList;
-import java.util.List;
 
 public class HomeView extends JPanel implements PropertyChangeListener{
     public final String viewName = "Home";
     HomeViewModel homeViewModel;
-    ArticleRetrievalController articleRetrievalController;
+    private ArticleRetrievalController articleRetrievalController;
```
Comment (Jaiz183) [inline src/view/HomeView.java:19]: Good job with noticing the non-private instance attribute!

**Pull request closed without merging**